app/src/main/java/com/example/felipe/criminalintent/CrimeFragment.kt

```diff
@@ -30,20 +26,21 @@ class CrimeFragment() : Fragment() {
             }
 
             btnCrime.text = crime.date.toString()
-            etxtTitle.addTextChangedListener(object : TextWatcher {
-                override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
-                    // This space intentionally left blank
-                }
-
-                override fun onTextChanged(
-                        s: CharSequence?, start: Int, before: Int, count: Int) {
-                    crime.title = s.toString()
-                }
-
-                override fun afterTextChanged(s: Editable?) {
-                    // This one too
-                }
-            })
+            editTxtTitle.addTextChangedListener { s, _, _, _ ->
+                crime.title = s
+            }
         }
     }
+
+    private fun EditText.addTextChangedListener(after: (s: Editable) -> Unit = {},
```

Comment:
what do you think of change this extension to use only on after text change? so you will have only one callback

Reply:
and change the name